modules/grandpa/src/benchmarking.rs

```diff
@@ -0,0 +1,210 @@
+	// We do this by creating many forks, whose head will be used as a signed pre-commit in the
+	// final justification.
+	submit_finality_proof_on_many_forks {
+		let n in 1..2;
+		let caller: T::AccountId = whitelisted_caller();
+
+		let init_data = InitializationData {
+			header: T::bridged_header(Zero::zero()),
+			authority_list: authority_list(),
+			set_id: TEST_GRANDPA_SET_ID,
+			is_halted: false,
+		};
+
+		initialize_bridge::<T, I>(init_data);
+
+		let mut header = T::bridged_header(One::one());
+		header.set_parent_hash(*T::bridged_header(Zero::zero()).parent_hash());
+
+		// TODO: Can't have more forks than we have authorities
+		let params = JustificationGeneratorParams {
+			header: header.clone(),
+			round: TEST_GRANDPA_ROUND,
+			set_id: TEST_GRANDPA_SET_ID,
+			authorities: keyring(),
+			depth: 1, // Maybe do 2?
+			forks: n,
+		};
+
+		let justification = make_justification_for_header(params).encode();
+
+	}: submit_finality_proof(RawOrigin::Signed(caller), header, justification)
+	verify {
+		assert!(true)
+	}
+
+
+	// Here we want to find out what the overheader of looking for an enacting an authority set is.
+	// I think we can combine the two benchmarks below into this single one...
+	// enacts_authority_set  {
+	// 	todo!()
+	// }: try_enact_authority_change(header)
+	// verify {
+	// 	assert!(true)
+	// }
+
+	// Here we want to find out the overheaded of looking through consensus digests found in a
+	// header.
+	//
+	// E.g, as the number of logs in a header grows, how much more work do we require to look
+	// through them?
+	//
+	// Note that this should be the same for looking through scheduled changes and forces changes,
+	// which is why we only have one benchmark for this.
+	find_scheduled_change {
+		// Not really sure what a good bound for this is.
+		let n in 1..1000;
+
+		let mut logs = vec![];
+		for i in 0..n {
+			// We chose a non-consensus log on purpose since that way we have to look through all
+			// the logs in the header
+			logs.push(sp_runtime::DigestItem::Other(vec![]));
+		}
+
+		let mut header = T::bridged_header(Zero::zero());
+		let digest = header.digest_mut();
+		*digest = sp_runtime::Digest {
+			logs,
+		};
+
+	}: {
+		crate::find_scheduled_change(&header)
+	}
+
+	// What we want to check here is how long it takes to read and write the authority set tracked
+	// by the pallet as the number of authorities grows.
+	read_write_authority_sets {
+		// The current max target number of validators on Polkadot/Kusama
+		let n in 1..1000;
+
+		let mut authorities = vec![];
+		for i in 0..n {
+			authorities.push((Alice, 1));
+		}
+
+		let authority_set = bp_header_chain::AuthoritySet {
+			authorities: authorities.iter().map(|(id, w)| (AuthorityId::from(*id), *w)).collect(),
+			set_id: 0
+		};
+
+		<CurrentAuthoritySet<T, I>>::put(&authority_set);
+
+	}: {
+		let authority_set = <CurrentAuthoritySet<T, I>>::get();
+		<CurrentAuthoritySet<T, I>>::put(&authority_set);
+	}
+}
+
+#[cfg(test)]
```

Comment:
Iiuc benchmarks tests won't run on CI. Do you think we still need it?

Reply:
(also seems that the tests are actually `assert(true)`)

Reply:
> Iiuc benchmarks tests won't run on CI. Do you think we still need it?

We should be running them on the CI, even if it's just periodically. I think that'll be a change for a different PR though.

I've found the tests to be pretty useful even if they're kinda "dumb" because they've helped me figure out if I've been setting up my benchmarks correctly (e.g are my justifications actually valid)